porch/pkg/registry/porch/secret.go

```diff
@@ -209,3 +197,24 @@ func (g *GcloudWIResolver) lookupGSAEmail(ctx context.Context, name, namespace s
 	}
 	return gsa, nil
 }
+
+type GcloudWICredential struct {
+	token *oauth2.Token
+}
+
+var _ repository.Credential = &GcloudWICredential{}
+
+func (b *GcloudWICredential) Valid() bool {
+	timeLeft := time.Until(b.token.Expiry)
+	if timeLeft > time.Minute {
```

Comment:
How long to these tokens typically last? If they are long I don’t see why we shouldn’t give ourselves more like an hour here. One minute doesn’t give enough time for retries on some temporary failure.

Reply:
The tokens seems to last for an hour from what I can see. 

I did originally use five minutes here, but I couldn't think of any reason why we would need such a long grace period. For retries we will do this check for every attempt, so it seems like it is mostly time diffs that could potentially cause a problem here.

I'm ok with increasing the value somewhat, I'm just not sure if it is needed.